bitwindow/lib/widgets/create_multisig_modal.dart

```diff
@@ -21,162 +23,289 @@ class CreateMultisigModal extends StatelessWidget {
+            SailButton(
+              label: 'Use My Key',
+              onPressed: () async => await model.useMyKey(),
+              variant: ButtonVariant.secondary,
+            ),
+            SizedBox(
+              width: 120,
+              child: Theme(
+                data: Theme.of(context).copyWith(
+                  textSelectionTheme: TextSelectionThemeData(
+                    selectionColor: theme.colors.primary.withValues(alpha: 0.2),
                   ),
-                  const SailSpacing(SailStyleValues.padding16),
-
-                  // Additional keys
-                  ...List.generate(
-                    model.m - 1,
-                    (i) => Padding(
-                      padding: const EdgeInsets.only(bottom: SailStyleValues.padding16),
-                      child: SailColumn(
-                        spacing: SailStyleValues.padding08,
-                        crossAxisAlignment: CrossAxisAlignment.start,
-                        children: [
-                          SailText.primary13('Key ${i + 2}'),
-                          SailTextField(
-                            controller: model.additionalKeyControllers[i],
-                            hintText: 'Enter public key',
-                          ),
-                        ],
-                      ),
+                ),
+                child: TextField(
+                  controller: model.keyNameController,
+                  onChanged: model.validateKeyNameFormat,
+                  maxLines: 1,
+                  decoration: InputDecoration(
+                    isDense: true,
+                    errorText: model.keyNameError,
+                    errorBorder: OutlineInputBorder(
+                      borderRadius: SailStyleValues.borderRadius,
+                      borderSide: BorderSide(color: theme.colors.error),
+                    ),
+                    focusedErrorBorder: OutlineInputBorder(
+                      borderRadius: SailStyleValues.borderRadius,
+                      borderSide: BorderSide(color: theme.colors.error),
+                    ),
+                    enabledBorder: OutlineInputBorder(
+                      borderRadius: SailStyleValues.borderRadius,
+                      borderSide: BorderSide(color: theme.colors.border),
+                    ),
+                    disabledBorder: InputBorder.none,
+                    focusedBorder: OutlineInputBorder(
+                      borderRadius: SailStyleValues.borderRadius,
+                      borderSide: BorderSide(color: theme.colors.text),
+                    ),
+                    hintText: 'Key Name',
+                    fillColor: theme.colors.background,
+                    filled: true,
+                    contentPadding: const EdgeInsets.symmetric(vertical: 11.5, horizontal: 12),
+                    hintStyle: SailStyleValues.thirteen.copyWith(
+                      color: theme.colors.inactiveNavText,
+                      fontSize: 12.0,
```

Comment:
Should not need to create your own TextField. Already have a SailTextField that should work? If not, what's wrong with it?